### PR TITLE
Request presentation

### PR DESCRIPTION
**Pull request by m00sey**
Before I go crazy testing everything wanted a review on this implementation.

I tried to isolate `request_presentations~attach` as that differs between indy/dif/jsonlds

### CODE REVIEW

File: pkg/static/canis-apiserver_swagger.go [hunk 1]
```diff
@@ -25,8 +25,8 @@ var staticFiles = map[string]*staticFilesFile{
...
 		hash:  "1ba6bd76284daab0cd648871a9205210238ab8e70e0d336cf8b70ac86593b864",
-		mime:  "application/json",
-		mtime: time.Unix(1601060386, 0),
+		mime:  "",
```
Comment (m00sey) [inline pkg/static/canis-apiserver_swagger.go:28]: so didn't mean to do this?

**Comment by m00sey** [comment]
> Enhance data model and datastore APIs for persisting needed data out of proof presentation.

blek this is missing

File: docker/canis/Dockerfile [hunk 2]
```diff
@@ -2,6 +2,8 @@ FROM scoir/indy-libs:latest
 
 #RUN apt-get update
 RUN apt-get install -y ca-certificates
+RUN apt-get update
+#RUN apt-get install -y ca-certificates
```
Comment (pfeairheller) [inline docker/canis/Dockerfile:6]: we should get rid of the commented lines.

File: pkg/apiserver/admin-presentproof.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+package apiserver
+
+import (
+	"context"
+	"github.com/scoir/canis/pkg/apiserver/api"
+)
+
+func (r *APIServer) RequestPresentation(ctx context.Context, request *api.RequestPresentationRequest) (*api.RequestPresentationResponse, error) {
+
+	panic("implement me")
```
Comment (pfeairheller) [inline pkg/apiserver/admin-presentproof.go:16]: Isn't this the last piece to tying it all together?

File: pkg/presentproof/engine/jsonld/engine.go [hunk 4]
```diff
@@ -3,13 +3,61 @@ package jsonld
...
+// RequestPresentation
+func (r *Engine) RequestPresentationAttach(attrInfo map[string]*api.AttrInfo, predicateInfo map[string]*api.PredicateInfo) ([]byte, error) {
+	log.Println(attrInfo)
+	log.Println(predicateInfo)
+
+	return nil, nil
+}
+
+func (r *Engine) RequestPresentationFormat() string {
+	return "??"
```
Comment (pfeairheller) [inline pkg/presentproof/engine/jsonld/engine.go:62]: LDS/LD-Proof?

**Review by pfeairheller — APPROVED** [review]

File: deploy/canis-chart/templates/apiserver-configmap.yaml [hunk 5]
```diff
@@ -62,6 +62,16 @@ data:
...
         host: canis-didcomm-issuer
         port: {{ .Values.issuer.grpc.port }}
+
+    ###############################################################
+    #
+    #  Verifier configuration
+    #
+    ###############################################################
+    issuer:
+      grpc:
```
Comment (pfeairheller) [inline deploy/canis-chart/templates/apiserver-configmap.yaml:72]: This needs to say "verifier"

File: pkg/presentproof/engine/indy/engine.go [hunk 6]
```diff
@@ -53,16 +54,65 @@ func New(prov provider) (*PresentationEngine, error) {
...
+		Nonce:               nonce,
+		RequestedAttributes: attrInfo,
+		RequestedPredicates: predicateInfo,
+	})
+	if err != nil {
+		return nil, err
+	}
 
-	return eng, nil
+	return b, nil
```
Comment (pfeairheller) [inline pkg/presentproof/engine/indy/engine.go:100]: We either need to base64 encode the return here... (see below)

File: pkg/presentproof/engine/registry.go [hunk 7]
```diff
@@ -36,6 +48,41 @@ func New(prov provider, opts ...Option) *Registry {
...
+			AttachID: attachID,
+			Format:   e.RequestPresentationFormat(),
+		}},
+		RequestPresentationsAttach: []decorator.Attachment{
+			{
+				ID:       attachID,
+				MimeType: "application/json",
+				Data: decorator.AttachmentData{
+					Base64: string(attach),
+				},
```
Comment (pfeairheller) [inline pkg/presentproof/engine/registry.go:78]: Or base64 encode what we get here...

**Review by pfeairheller — CHANGES_REQUESTED** [review]
Sweet stuff, only 2 changes needed.

**Review by pfeairheller — APPROVED** [review]

**Pull request merged**